### PR TITLE
Allow password enabled for iso (#2745)

### PR DESCRIPTION
**Pull request by anuragaw**
Both register and update iso commands can allow for allowing password enabled.

Fixes: #2745 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Ran register iso and update iso API through cloudmonkey. Also added passwordenabled field changes to true after update iso api call through test_iso.py#test_02_edit_iso

### CODE REVIEW

File: api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java [hunk 1]
```diff
@@ -177,6 +182,10 @@ public boolean isDirectDownload() {
         return directDownload == null ? false : directDownload;
     }
 
+    public Boolean isPasswordEnabled() {
+        return passwordEnabled;
```
Comment (rhtyd) [inline api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: use a ternary operator like, `var == null ? false : var`

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by anuragaw** [comment]
@blueorangutan help

File: api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java (hunk 1 above)
Comment (anuragaw) [reply api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: Thanks for the reviews @rhtyd , @rafaelweingartner. I'm fixing this and pushing a new commit.

Comment (rhtyd) [reply api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: @anuragaw don't see your new commit yet

**Comment by anuragaw** [comment]
@blueorangutan package

**Comment by anuragaw** [comment]
@blueorangutan test

File: api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java (hunk 1 above)
Comment (anuragaw) [reply api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: Apologies for the delay @rhtyd . I have pushed another commit and ran a test with blueorangutan.

File: api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java [hunk 2]
```diff
@@ -113,6 +113,11 @@
             description = "true if ISO should bypass Secondary Storage and be downloaded to Primary Storage on deployment")
     private Boolean directDownload;
 
+    @Parameter(name = ApiConstants.PASSWORD_ENABLED,
+            type = CommandType.BOOLEAN,
+            description = "true if password reset feature is supported; default is false")
+    private Boolean passwordEnabled;
```
Comment (rhtyd) [inline api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: @anuragaw let's export this via the template response (listIsos response should return this key) and we're golden!

Comment (anuragaw) [reply api/src/main/java/org/apache/cloudstack/api/command/user/iso/RegisterIsoCmd.java]: Added this in the 3rd commit @rhtyd . This was also prohibiting adding tests for update iso cmd in test_iso which works well now. Thanks!

File: test/integration/smoke/test_iso.py [hunk 3]
```diff
@@ -332,6 +333,13 @@ def test_02_edit_iso(self):
             self.services["ostypeid"],
             "Check OSTypeID of updated ISO"
         )
+
+        self.assertEqual(
+            iso_response.passwordenabled,
+            self.services["passwordenabled"],
```
[the remark this answers is not in the document]
Comment (anuragaw) [reply test/integration/smoke/test_iso.py]: Yes @rhtyd . The default configs in test_data has false passwordenabled for all templates/iso. The global passwordenabled key is true and tests false->true field change test for templates/Isos.

**Comment by anuragaw** [comment]
> @anuragaw please put a Fixes #xyz ID in the description.

Done @rhtyd

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by anuragaw** [comment]
Test "test_02_edit_iso" was failing because local test_data file for Marvin tests uses Bool value in JSON format but the Trillian Environment uses String to represent "True", "False". This requires us to typecast config values for comparison.

**Comment by anuragaw** [comment]
Jekins build failure is from surefire plugin similar to what we have for building CloudStack in Debian environment (from ML) -
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.21.0:test (default-test) on project cloud-utils: There are test failures.`

**Comment by anuragaw** [comment]
@blueorangutan package

**Comment by anuragaw** [comment]
@blueorangutan test

**Pull request merged**